src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+#include <string>
+#include <vector>
+#include <SFML/Graphics/Color.hpp>
+#include <array>
+#include <cmath>
+#include "ColorMap.h"
+#include <iostream>
+
+using namespace std;
+using namespace sf;
+
+/**
+ * This Method will compute in terms of the argument sequences and the k, all kmers we are looking for and call
+ * the method which compute us in terms of this number a Colormap. Then the method will compute
+ * all different Kmers and push them in a List. We'll use this list to initialize differentKmers.
+ */
+ColorMap::ColorMap(vector<string>& Sequences, int k){
+    ///< This is a variable which save the number of Kmers per Sequence which we have
+    vector<int> SequenceSizes;
+    for (int i=0;i<Sequences.size();i++)
+        SequenceSizes[i] = (int)floor((Sequences[i].size())/k);
+
+    ///< This is a variable which save the number of Kmers
+    int numbOfKmer = 0;
+    for (int i=0;i<SequenceSizes.size();i++)
+        numbOfKmer += SequenceSizes[i];
+    updateColorList(numbOfKmer);
+    ///< This variable is the output List
+    vector<string> kmerDeclarations;
+
+    ///< a boolean for the "got this kmer a color?"-Check
+    bool matched = false;
+    ///< a index for the k-mers per sequence
+    int index;
+    
+    /**
+     * This part of the method will push all Kmers in the List, which are not in the list already.
+     */
+    for (int i=0;i<Sequences.size();i++) {
+        index = 0;
+        while (index<SequenceSizes[i]) {
+            ///< Comparing all k-mers to know if the Kmer is already in the list
+            for (int j=0;j<kmerDeclarations.size();j++) {
+                if (kmerDeclarations[j]==Sequences[i].substr(index*k, k)) {
+                    matched = true;
+                    break;
+                }
+            }
+            ///< If its a new kmer, so we push it here
+            if (!matched)
+                kmerDeclarations.push_back(Sequences[i].substr(index*k, k));
+            matched = false;
+            index += 1;
+        }
+    }
+    differentKmers = kmerDeclarations;
+}
+
+/**
+ * This Method will make a vector of Colors in terms of a number, which defines the Length of the 
+ * length of the List. Then it will initialize ColorList with it. In our case the argument 
+ * describes the number of Kmers we are looking for to map.
+ */
+ColorMap::ColorMap(int numbOfKmer){
+    vector<Color> returnList; ///< the list i use to adjust this as the new List
+    Color placeholder; ///< A placeholder which is needed to programm with rgb values
+    /**
+     * In this part of the method i push Colors in the List which i searched in the
+     * web, so that they are well different.
+     */
+    placeholder.r = 255;
```

Comment:
Do not encode data like this. Suggestion:
```
unsigned char someDistinguishableCols[10][3] = { 
    {255,0,0}, // red
    {0,255,0}, // green
    {0,0,255}, // blue
...
};
```
A longer or shorter vector of Colors can then be constructed using this array.

Reply:
Also, include the web link as a comment in the code when you found such a list of colors online.